docs/admin/Offline_System_Certificate_Renewal.md

```diff
@@ -6,71 +6,105 @@ Offline System Certificate Renewal
+3. Run the following command providing **all** the certs that needs to be renewed. Command should complete without error.
 
     ````
-    # certutil -L \
-    -d <admin NSS database> \
-    -n <admin cert nickname>
+    # pki-server cert-fix \
+    --ldapi-socket /var/run/slapd-REALM.socket \
+    --agent-uid admin \
+    --cert sslserver \
+    --cert subsystem \
+    --cert ca_ocsp_signing \
+    --cert ca_audit_signing \
+    --extra-cert $IPA_RA_SERIAL \
+    --extra-cert $DS_SERIAL \
+    --extra-cert $HTTPD_SERIAL
+    ````
+
+4. Verify that there is no `internaldb` field in `/etc/pki/pki-tomcat/password.conf`
+
+5. Verify that `CS.cfg` has `internaldb.ldapauth.authtype=SslClientAuth`
 
-    Certificate:
-        Data:
-            Version: 3 (0x2)
-            Serial Number: 6 (0x6)
-            Signature Algorithm: PKCS #1 SHA-256 With RSA Encryption
-            Issuer: "CN=CA Signing Certificate,OU=pki-tomcat,O=EXAMPLE"
-            Validity:
-                Not Before: Sat Dec 15 02:16:26 2018
-                Not After : Fri Dec 04 02:16:26 2020
-            Subject: "CN=PKI Administrator,E=caadmin@example.com,OU=pki-tomcat,O=
-                EXAMPLE"
+6. Copy `/etc/pki/pki-tomcat/certs/$IPA_RA_SERIAL-renewed.crt` to `/var/lib/ipa/ra-agent.pem`
 
+7. Remove old DS cert and import DS renewed cert into DS NSSDB and provide the pin from `/etc/dirsrv/slapd-REALM/pin.txt`
     ````
+    # certutil -D -n Server-Cert \
+    -d /etc/dirsrv/slapd-REALM/
 
-## Automated Renewal Process
+    # certutil -A -n Server-Cert \
+    -d /etc/dirsrv/slapd-REALM/ \
+    -t ',,' \
+    -a -i /etc/pki/pki-tomcat/certs/${DS_SERIAL}-renewed.crt
+    ````
+8. Copy `/etc/pki/pki-tomcat/certs/$HTTPD_SERIAL` to `/var/lib/ipa/certs/httpd.crt`
 
-One line tool that fixes all certificates:
+9. `ipactl restart` should succeed
 
-    # pki-server cert-fix \
-    -n <admin cert nickname> \
-    -d <admin NSS database> \
-    -c <admin NSS database password>
+10. Verify that PKI operations succeed
+    ````
+    # pki -U https://<host>:8443 ca-cert-find
+    ````
 
-One line tool to fix one particular certificate:
+### Standalone PKI environment (Uses LDAPS)
 
-    # pki-server cert-fix --cert <cert ID> \
-    -n <admin cert nickname> \
-    -d <admin NSS database> \
-    -c <admin NSS database password>
+#### Assumptions:
+
+- Valid CA certificate
+- TLS configured Directory Server
+- If Dogtag was configured to use TLS certificate authentication to bind to LDAP, a Valid DS service certificate
+
+#### Usage:
+
+One line tool that fixes all certificates:
+
+    # pki-server cert-fix \
+    --ldap-url <LDAP URL> \
+    --agent-uid <agent UID>
 
 For all available options, you can type:
 
     $ pki-server cert-fix --help
 
 ## Manual Renewal Process
+
+**NOTE:** The steps listed here are for a *PKI standalone environment*. For the *IPA Environment*, the steps are very complicated and it's
+suggested to use the [automated process](#ipa-environment-uses-ldapi)
```

Comment:
Probably worth mentioning that IPA now has its own wrapper tool, `ipa-cert-fix` to simplify the process.  It uses `pki-server cert-fix` behind the scenes.